aws-lc-rs/src/aead.rs

```diff
@@ -631,6 +689,37 @@ impl LessSafeKey {
         seal_in_place_separate_tag_(&self.key, nonce, Aad::from(aad.as_ref()), in_out)
     }
 
+    /// Like `SealingKey::seal_in_place_scatter()`, except it accepts an
+    /// arbitrary nonce.
+    ///
+    /// `nonce` must be unique for every use of the key to seal data.
+    ///
+    /// # Errors
+    /// `error::Unspecified` if encryption operation fails.
+    ///
+    #[inline]
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn seal_in_place_scatter<A>(
+        &self,
+        nonce: Nonce,
+        aad: Aad<A>,
+        in_out: &mut [u8],
+        extra_in: &[u8],
+        extra_out_and_tag: &mut [u8],
```

Comment:
I'm trying to think how we would best support this in the "FIPS" API for AES-GCM, see [`TLSRecordSealingKey`](https://github.com/aws/aws-lc-rs/pull/207). The AWS-LC API writes the randomized nonce to the end of the tag (thus the output tag is 12-bytes longer). This to me is an implementation detail of AWS-LC (likely having been done by the BoringSSL project to add the random nonce support within the existing API constraints), and not one I wish to force have aws-lc-rs users to have to think about. I'd like for our users to see those tags as standard sizes, and we provide the generated nonce as part of the output of the function.

I raise this issue because the `extra_in` API would require us to ask the user to size the `extra_out_and_tag` slice appropriately with the additional padding to write a nonce, which would leak this aspect of how the underlying API works. If we wanted to accept leaking this detail for this API, we could document the behavior, and require customers provided a slice that is `extra_in.len() + tag_len + 12 bytes`, but then I think the API should return a proper reference to the slice in the output result that is only `extra_in.len() + tag_len`. So something like `Result<&mut [u8], Unspecified>`. Otherwise if we absolutely don't want to expose this detail to the user then we are going to need to size an internal buffer and perform a memcpy, which may negate the benefits at that point, even if the buffer would be located on the stack.